Temp and humidty sensor

lucasnot opened the pull request:
New pull request because I deleted the new one

lucasnot committed:
Update PSoCCreatorExportIDE.xml

lucasnot committed:
Added dht11.c/.h

Preliminary to making the sensors function

lucasnot committed:
Fixed some issues with merge

lucasnot committed:
Changed loop in dht11 and more

Changed the loop in dht11.c so that it does not run infinitly, instead,
ends when the two values are read in

lucasnot committed:
Finishing touches on this part

Formatting and what not

lucasnot committed:
Minor Fixes

Fixed a few things detailed in the pull request review

lucasnot committed:
Minor fixes and style fixes

lucasnot committed:
Added Calls in DHT.h and Adjusted the .c

lucasnot committed:
Minor Fixes

djpetti reviewed (changes requested):
I think you already have all my comments, Lucas.

djpetti commented:
This got closed when I deleted the psoc branch. All that code is in the devel branch now, so I've changed the base, and am reopening this. Incidentally, it will probably also have to be rebased.

lucasnot committed:
Switch cases

Put the loop that reads in the temperature and humidity in a switch
case. This was its not running awlays, and won't run unless we initilize
it again.

lucasnot committed:
Removed extrainous comments.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+_CY_ISR_PROTO(Delay_Timer_ISR_Handler);
+_CY_ISR_PROTO(DHT_Pin_ISR_Handler);
+
+// What Return_Decoded_Temperature does:
```
Update function name in comment.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+    
+    if(g_decoded_temperature_data == NULL) {
+        return -1;
+    }
```
Put else on same line as closing brace.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+    }    
+}    
+
+// What Return_Decoded_Humidity does:
```
Update function name in comment.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+
+uint8_t dht_get_temperature(){
+    
+    if(g_decoded_temperature_data == NULL) {
```
There should be a space between "if" and "(". Same for "else" and "{".

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+    g_executed = 0;
+    Delay_Timer_ISR_Enable();
+    // Write 0 to Control_Reg_Delay_Timer to starting Delay_Timer component 
+    Control_Reg_Delay_Timer_Write(0);
```
Are you sure there's not just a ```Delay_Timer_Start()``` function you can call?

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+
+
+// What decode dht data does:
+//    - The value stored in g_countvalues_datatimer[] array is used to decode
```
I would explain what the function's for a little more abstractly. If we wanted that much detail, we can always look at your code. Remember, the comment blocks are here so that I can glance over your function and get a rough idea of what it does, etc. If you need guidance, look at the comments I've put on some of my functions.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+// Parameters: None
+// Returns: None
+
+void decode_dht_data(void){
```
Space before {

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+
+void decode_dht_data(void){
+    int i;
+    for(i=0; i<44; ++i) {
```
Spaces around "=" and "<". Also, what is 44? This should at least have a comment. Better yet, make it a named constant and stick it at the top of the file.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+    }
+    
+    int j;
+    for(j=0; j<8; ++j) {
```
Spaces around "=" and "<" again. Also, it looks like there are some more "magic" numbers here that need documentation.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+    DHT_Pin_ClearInterrupt();
+    // Read the counter values of Data_Timer for processing
+    g_countvalues_datatimer[g_risingedge_counts]=Data_Timer_ReadCounter();
+    g_risingedge_counts++;
```
Prefer preincrement when you can. It's more efficient.

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+    // Clear the interrupt request of the pin DHT_Pin
+    DHT_Pin_ClearInterrupt();
+    // Read the counter values of Data_Timer for processing
+    g_countvalues_datatimer[g_risingedge_counts]=Data_Timer_ReadCounter();
```
Space around "=".

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c:
```diff
@@ -0,0 +1,273 @@
+    // be in the respective variables
+    
+    /*
+    switch(g_has_been_executed){
```
Everything below here looks like the old blocking implementation. Did you mean to get rid of it?

djpetti commented on psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.h:
```diff
@@ -0,0 +1,23 @@
+#include <stdint.h>
+    
+    
+//Initilize everything needed
```
Space after //

djpetti reviewed (changes requested):
The interrupt-driven stuff is moving in the right direction.

Also, you need to rebase this on devel before it can get merged.

lucasnot committed:
Stopped infinite loop

This change makes it so that instead of the temp and humidity sensor
being continually updated, it doesnt update until someone calls it agian

lucasnot committed:
Same as before

I guess this didn't go with the previous commit

lucasnot committed:
Stylistic Changes

Just changing things to be more pleasing to look at and easier to read.

lucasnot committed:
Man I'm bad at this

lucasnot committed:
Added documentation where needed

lucasnot committed:
Changed three seperate if statements to else if

This change is for increased functionality and also a sense of
cleanliness in the code

lucasnot committed:
Took away the switch

Not needed when the functions are called properly

lucasnot committed:
Removed extranious variable

We don't need it any more, since we took away the switch statment

djpetti commented:
Have you tested this on actual hardware yet?
